src/plugins/data/server/index_patterns/deprications/scripted_fields.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  CoreSetup,
+  DeprecationsDetails,
+  GetDeprecationsContext,
+  RegisterDeprecationsConfig,
+} from 'kibana/server';
+import { IndexPatternAttributes } from '../../../common';
+
+type IndexPatternAttributesWithFields = Pick<IndexPatternAttributes, 'title' | 'fields'>;
+
+export const createScriptedFieldsDeprecationsConfig: (
+  core: CoreSetup
+) => RegisterDeprecationsConfig = (core: CoreSetup) => ({
+  getDeprecations: async (context: GetDeprecationsContext): Promise<DeprecationsDetails[]> => {
+    const finder = context.savedObjectsClient.createPointInTimeFinder<IndexPatternAttributesWithFields>(
+      {
+        type: 'index-pattern',
+        perPage: 100,
+        fields: ['title', 'fields'],
+      }
+    );
```

Comment:
@Bamieh do you think we should introduce some kind of caching in the deprecation service? atm everytime the `/api/deprecations` route is called, all the deprecation providers are executed, and I wonder if this will be alright when we'll have a lot that are performing ES requests like this one. wdyt?

Reply:
If we do, then need to be careful? It is likely that with the cashing in place the use case I expressed in e2e testtest/api_integration/apis/index_patterns/deprecations/scripted_fields.ts might break?

Reply:
@Dosant this should not impact this PR, I was more wondering about long-term performances.

Reply:
Initially i had caching in place for the service but it didnt make sense since to keep it. The API is called only at the user's request when they open the UA or explicitly via the API. Generic caching wont work since we do not want to return already resolved deprecations from the api. We always want the latest state of the deprecations.

For performance it might make sense to introduce pagination rather than caching. The service would paginate the kibana deprecations so we dont have to loop over all deprecations every time. Since the deprecations are registered at setup in an array we can guarantee the order and create basic pagination from the public contract.

I agree with @pgayvallet this doesnt impact the PR. I'll open a separate issue to continue the discussion there.